ex2/ex2.js

```diff
@@ -8,3 +8,24 @@
  *     fn(",,,", 4) === -1
  *     fn("hel,lowo,rld", 0) - Throws
  */
+function getTheCommaIndexAt(str, commaIndex) {
+
+	if (commaIndex === 0) {
+		throw new RangeError('Throws...');
+	}
+
+	var index = -1;
+
+	for (var i = 0; i < commaIndex; i++) {
+
+		index = str.indexOf(',',index + 1);
+
+		//  If indexOf return -1 no comma found so break and return -1
+		//  If it is the commaIndex value and greater than -1
+		if (i === (commaIndex - 1) && index > -1 || index === -1) {
```

Comment:
When index===-1, the entire condition is true because of the second part. In this case, the `&& index > -1` part doesn't matter at all.
We can write this as a table (search for truth table: https://www.google.com/search?q=logic+truth+tables&espv=2&source=lnms&tbm=vid&sa=X&ved=0ahUKEwjyj8vUyNPQAhUUSGMKHWgCApsQ_AUICigD&biw=1411&bih=748&dpr=0.9):


```
  i===commaIndex - 1  |   index || condition
 =====================|=========||===========
       true           |   -1    ||    true
       true           |   > -1  ||    true
       false          |   -1    ||    true
       false          |   > -1  ||    false
```

And this is a simple or:
`if (i === (commaIndex - 1) || index === -1) `